virtool/flags.py

```diff
@@ -1,46 +1,50 @@
-from typing import List, Callable, Dict
+from typing import List, Callable
 from enum import Enum
 from aiohttp import web
 from aiohttp.web_exceptions import HTTPNotFound
 from aiohttp.web import Request
 
-import virtool.http.routes
-from virtool.types import App
-
-routes = virtool.http.routes.Routes()
-
 
 class FlagName(Enum):
-    FF_SPACES = "FF_SPACES"
-    FF_ML_MODELS = "FF_ML_MODELS"
-    FF_ADMINISTRATOR_ROLES = "FF_ADMINISTRATOR_ROLES "
+    SPACES = "SPACES"
+    ML_MODELS = "ML_MODELS"
+    ADMINISTRATOR_ROLES = "ADMINISTRATOR_ROLES"
 
 
 class FeatureFlags:
     """
-    Class contains the derived value of the feature flags and cli flag config.
+    Class contains the derived value of the feature flags.
     """
 
-    def __init__(self):
+    def __init__(self, cli_flags: List[FlagName]):
         self.constants = {
-            FlagName.FF_ADMINISTRATOR_ROLES.value: True,
-            FlagName.FF_ML_MODELS.value: False,
-            FlagName.FF_SPACES.value: False,
+            FlagName.ADMINISTRATOR_ROLES: True,
+            FlagName.ML_MODELS: False,
+            FlagName.SPACES: False,
         }
+        self.set_derived_flag_values(cli_flags)
 
-    def derive_flag_values(self, cli_flags: List[str]) -> Dict[str, bool]:
+    def set_derived_flag_values(self, cli_flags: List[FlagName]):
         """
-        Derives flag values based on the hardcoded values and the cli flags.
+        :param cli_flags:
         """
-        return {
-            constant_name: constant_value if constant_name not in cli_flags else True
-            for constant_name, constant_value in self.constants.items()
-        }
+        for attr_name, constant_value in self.constants.items():
+            setattr(
+                self,
+                attr_name.name,
+                constant_value if attr_name not in cli_flags else True,
+            )
+
+    def check_flag_enabled(self, feature_flag: FlagName):
+        if feature_flag is None:
+            return None
+        return getattr(self, feature_flag.name)
 
 
-def flag(feature_flag: str):
+def flag(feature_flag: FlagName):
     """
     Sets the feature flag attribute for the route.
+    :param feature_flag: feature flag name associated with the route
```

Comment:
This docstring is not very helpful.

Imagine someone wants to use this decorator. First, they can't easily know it is a decorator unless it is mentioned in the docstring. Then, the docstring just tells them some low-level detail instead of what the function will actually accomplish for them.

When writing docstrings, focus on 

```suggestion
    Prevents access to the decorated request handler then the passed ``feature_flag`` is not enabled.
    
    :param feature_flag: feature flag name associated with the route
```